### ISSUE TITLE
New testcloud version throws an exception

### ISSUE THREAD

**Issue by FrNecas** [psss/tmt#837]
Reproducer with the latest tmt from master branch (with testcloud 0.6.0):
```
mkdir test && cd test
tmt init && tmt plan create -t mini plan
tmt run
```

results in:
```
/var/tmp/tmt/run-056

/plan
    discover
        how: shell
        directory: /home/fnecas/devel/tmttest/test
        summary: 1 test selected
    provision
        how: virtual
        image: fedora
        user: root
        memory: 2048 MB
        disk: 10 GB
        progress: booting...
    finish
        summary: 0 tasks completed
Traceback (most recent call last):
  File "/home/fnecas/.virtualenvs/tmt/bin/tmt", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/fnecas/devel/tmt/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1647, in invoke
    return _process_result([] if self.chain else None)
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1637, in _process_result
    value = ctx.invoke(self._result_callback, value, **ctx.params)
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/fnecas/devel/tmt/tmt/cli.py", line 300, in finito
    click_context.obj.run.go()
  File "/home/fnecas/devel/tmt/tmt/base.py", line 1404, in go
    plan.go()
  File "/home/fnecas/devel/tmt/tmt/base.py", line 765, in go
    step.go()
  File "/home/fnecas/devel/tmt/tmt/steps/provision/__init__.py", line 95, in go
    plugin.go()
  File "/home/fnecas/devel/tmt/tmt/steps/provision/testcloud.py", line 237, in go
    self._guest.start()
  File "/home/fnecas/devel/tmt/tmt/steps/provision/testcloud.py", line 458, in start
    self.instance.spawn_vm()
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/testcloud/instance.py", line 613, in spawn_vm
    self.write_domain_xml()
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/testcloud/instance.py", line 544, in write_domain_xml
    lock = util.Filelock()
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/testcloud/util.py", line 165, in __init__
    self.fd = open(lock_path, 'w+')
PermissionError: [Errno 13] Permission denied: '/var/lib/testcloud/testcloud.lock'
Exception ignored in: <function Filelock.__del__ at 0x7f4beff908b0>
Traceback (most recent call last):
  File "/home/fnecas/.virtualenvs/tmt/lib/python3.9/site-packages/testcloud/util.py", line 193, in __del__
    self.fd.close()
AttributeError: 'Filelock' object has no attribute 'fd'
```

Looks like testcloud is trying to create a lock `/var/lib/testcloud/testcloud.lock` (but without root it doesn't have permission to write there), it fails and hence the `fd` attribute is not set. The destructor then assumes it is set which is not true. Looks like a bug in testcloud (CC: @frantisekz ). Can we do anything about it from the side of tmt?

**Comment by frantisekz** [comment]
@FrNecas thanks for reporting the issue. I've prepared a PR which fixes the behavior for me: https://pagure.io/testcloud/pull-request/113 

The problem was with using default arg in FileLock class. Having it in a separate file which is importing testcloud's config caused that it didn't use the value provided by tmt.  Also, it seems only non-rpm testcloud installations were affected (eg. from pypi), I'll keep in mind that I need to test these before making a next release.

Can you somehow give it a try? Thanks!

**Comment by frantisekz** [comment]
If it doesn't hurry too much, I'll cut next minor testcloud release either on Monday or Tuesday. I can fast-track this and release today if necessary.

**Comment by FrNecas** [comment]
@frantisekz Thanks a lot for the quick fix! I just gave it a try on your branch with tmt and it works great. There is no need to rush, Monday or Tuesday works fine. Tmt's release is planned for the end of the month so it would be great to have it fixed for the release. I will keep the issue open for now, since we will have to bump the required version. Can you please let us know once the release is ready for use? Thanks!

**Comment by frantisekz** [comment]
testcloud 0.6.1 with fix for this has been released, is available on pypi and pending testing for f33, f34, epel8.

**Comment by psss** [comment]
Thanks for the fix. Tested a bit and everything looks ok. I've also created a basic [integration test](https://pagure.io/testcloud/pull-request/114) which can be used to verify that `tmt` with `testcloud` works fine across distros. Needs nested virtualization though, otherwise we could enable it in the Fedora CI as well.

@thrix, btw, do we have a plan to support nested virtualization in Testing Farm / Fedora CI?
